functiondefextractor/core_extractor.py

```diff
@@ -96,123 +262,379 @@ def get_func_body(filename, line_num):
+        @parameters
+        code_list: list to store the extracted methods
+        line_num: list of function line numbers
+        func_name: list of function names
+        annot: given annotation condition (Ex: @staticmethod)
+        @return
+        This function returns extracted python methods"""
+    for lin_no, func in zip(line_num, functions):
+        if check_annot(func_name, lin_no, annot) is not None:
+            code_list.append(check_annot(func_name, lin_no, annot))
+            UID_LIST.append(func_name + "_" + func)
+    return code_list
+
+
+def get_final_dataframe(delta, code_list):
+    """ This function processes extracted data to generate a dataframe
+         @parameters
+        code_list: list of extracted methods
+        delta: Required lines from method
+        @return
+        This function returns a dataframe of extracted function/methods"""
+    if delta is not None:
+        ret_val = process_delta_lines_data()
+    else:
+        ret_val = process_final_data(code_list)
+    return ret_val
+
+
+def clean_log():
+    """ Function to clean the log file"""
+    ini_path = os.path.abspath(os.path.join
+                               (os.path.dirname(__file__), os.pardir))
+    file_name = os.path.join(ini_path, "functiondefextractor", "extractor.log")
+    if os.path.exists(file_name):
+        open(file_name, 'w').close()
+
+
+def get_log_data(line):
+    """ function to get the line requested from log data"""
+    ini_path = os.path.abspath(os.path.join
+                               (os.path.dirname(__file__), os.pardir))
+    file_name = os.path.join(ini_path, "functiondefextractor", "extractor.log")
+    file_variable = open(file_name, encoding='utf-8', errors='ignore')  # pragma: no mutate
+    all_lines_variable = file_variable.readlines()
+    string = all_lines_variable[-line]
+    string = string[0: 0:] + string[23 + 1::]
+    return string
+
+
+def remove_comments(dataframe):
+    """ This function removes comments from the code extracted
+            @parameters
+            dataframe: extracted methods in dataframe format
+            @return
+            This function returns function/method definitions by removing comments"""
+    filtered_code = []
+    data = ""
+    for i in range(len(dataframe).__trunc__()):
+        for line in dataframe.iat[i, 0].splitlines():
+            if not line.strip().startswith(("#", "//", "/*", "*", "*/")):  # pragma: no mutate
+                data = data + line.strip().split(";")[0] + os.linesep
+        filtered_code.append(data)
+        data = ""
+    dataframe["Code"] = filtered_code
+    return dataframe
+
+
+def get_report(data, path):
+    """ This function classifies the report files based on the file type(Ex: .java, .cs, .py, etc.)
+                @parameters
+                data: extracted methods in dataframe format
+                path: Report folder path"""
+    method_data = [[] for _ in range(7)]
```

Comment:
what is specific with 7?

Reply:
now functiondefextractor supports 7 programming languages